diff --git a/telegram-bot-api/Client.cpp b/telegram-bot-api/Client.cpp
index 83f88ae..ee7ca11 100644
--- a/telegram-bot-api/Client.cpp
+++ b/telegram-bot-api/Client.cpp
@@ -7902,16 +7902,15 @@ void Client::json_store_file(td::JsonObjectScope &object, const td_api::file *fi
   if (with_path && file->local_->is_downloading_completed_) {
     if (parameters_->local_mode_) {
       if (td::check_utf8(file->local_->path_)) {
-        object("file_path", file->local_->path_);
+        object("local_file_path", file->local_->path_);
       } else {
-        object("file_path", td::JsonRawString(file->local_->path_));
-      }
-    } else {
-      Slice relative_path = td::PathView::relative(file->local_->path_, absolute_dir_, true);
-      if (!relative_path.empty() && file->local_->downloaded_size_ <= MAX_DOWNLOAD_FILE_SIZE) {
-        object("file_path", relative_path);
+        object("local_file_path", td::JsonRawString(file->local_->path_));
       }
     }
+    Slice relative_path = td::PathView::relative(file->local_->path_, absolute_dir_, true);
+    if (!relative_path.empty() && file->local_->downloaded_size_ <= MAX_DOWNLOAD_FILE_SIZE) {
+      object("file_path", relative_path);
+    }
   }
 }
 
